### PR TITLE
[BUD-206] Contact Details Screen (Newbie)

### PR DESCRIPTION
**Pull request by michaelmoney**
# Feature name

# [BUD-206](https://digitalrig.atlassian.net/browse/BUD-206) Contact Details Screen (Newbie)

## Description
Component `<ContactDetails />` displays user contact details based on route. For example:

**Newbie Contact Details:**
URL: http://localhost:3000/newbie/buddy/ck17stp549j6q0b17i907lop9/details

**Buddy Contact Details:**
URL: http://localhost:3000/buddy/newbies/ck360gf1rppl70964ys3bbfdo/details

## Screenshots

newbie

buddy

### CODE REVIEW

File: src/components/ContactDetails/ContactDetails.tsx [hunk 1]
```diff
@@ -1,119 +1,81 @@
...
+      };
+    } else if (isNewbie(role)) {
+      return {
+        query: BUDDY_CONTACT_DETAILS,
+        variables: { buddyId },
+        userRole: UserRole.Buddy.toLowerCase(),
+      };
+    }
+  };
+  const { query, variables, userRole } = getQueryByRole(role) || {};
```
Comment (Tonik1204) [inline src/components/ContactDetails/ContactDetails.tsx]: As a suggestion - seems this part can be updated to use less code and resources:
```
const QUERY_BY_ROLE = {
  [UserRole.Buddy]: {
        query: BUDDY_CONTACT_DETAILS,
        variables: { buddyId },
        userRole: UserRole.Buddy.toLowerCase(),
      },
  [UserRole.Newbie]: {
        query: NEWBIE_CONTACT_DETAILS,
        variables: { newbieId },
        userRole: UserRole.Newbie.toLowerCase(),
      },
}; 
const { query, variables, userRole } = QUERY_BY_ROLE[role];
```

Comment (michaelmoney) [reply src/components/ContactDetails/ContactDetails.tsx]: Thanks @Tonik1204 for a tip :+1: . That's slightly cleaner approach. I've added it here: f7dd3b5188250d097bfebbf8063d64d015dce6b5

File: src/components/UserDetails/UserDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    phoneNumber,
+    notes,
+    role,
+  } = props.details;
+
+  const isNotesVisible = (role: UserRole) => isNewbie(role);
+
+  return (
+    <div data-testid='contact-details-page'>
+      <Box className={wrapper}>
```
Comment (lucianojss) [inline src/components/UserDetails/UserDetails.tsx]: You can remove this div and pass ` data-testid='contact-details-page' to Box component. And minimize DOM tree

Comment (michaelmoney) [reply src/components/UserDetails/UserDetails.tsx]: Done. 3176e1bb0d0a5f5a1a039bd0a390a5ec91925c79

**Pull request merged**